.pre-commit-config.yaml

```diff
@@ -32,6 +32,13 @@ repos:
     language: system
     types: [go]
 
+  - id: golangci-lint
```

Comment:
Just curious why we use a local script instead of upstream pre-commit hook -> https://github.com/golangci/golangci-lint/blob/master/.pre-commit-hooks.yaml.

I suppose it's to avoid syncing version /w Makefile? I personally like to throw all formatters/linters in pre-commit and let this be the authoritative source. And then just create a `make pre-commit` target that runs on all files.

This way you can avoid all those sub-targets in the Makefile (`goimports`, `govet`, `gofmt`, `golanci-lint`...)

In any case LGTM!

Reply:
TIL golangci-lint has a pre-commit hook as well! I just knew of the one in pre-commit-golang [1].

But both have the problem that they use the `golangci-lint` that is available in your `$PATH` and we want to use the version that gets installed via the Makefile.

Does that make sense or am I overlooking something?

[1] https://github.com/dnephin/pre-commit-golang

Reply:
Oh youre right, I thought it would pull golangci-lint with `go get ...` or something. Nevermind ;)